src/app/study-filters-block/study-filters-block.component.ts

```diff
@@ -19,29 +21,56 @@ export class StudyFiltersBlockComponent extends StatefulComponent implements OnI
 
   @Validate(SetNotEmpty, {message: 'Select at least one.'})
   public selectedStudies: Set<string> = new Set();
+  public allStudyNodes: DatasetNode[][];
 
   @ViewChild('nav') public ngbNav: NgbNav;
 
-  public constructor(protected store: Store) {
+  public constructor(protected store: Store, public datasets: DatasetsService) {
     super(store, StudyFiltersBlockState, 'studyFilters');
   }
 
   public ngOnInit(): void {
     super.ngOnInit();
-
-    this.allStudies = new Set<string>(this.dataset.studyNames);
-
+    this.allStudies = new Set<string>();
+    const datasets$ = this.datasets.getDatasetsObservable();
+    const datasetTrees = new Array<DatasetNode>();
+    datasets$.subscribe(datasets => {
+      datasets.filter(d => !d.parents.length).map(d => {
+        this.selectedStudies.add(d.id);
+        this.updateSelectedStudies(this.selectedStudies);
+        datasetTrees.push(new DatasetNode(d, datasets));
+      });
+      this.allStudyNodes = datasetTrees.map((dataset) =>
+        dataset.dataset.id !== this.dataset.id ?
+          dataset.children.filter(child => child.dataset.id === this.dataset.id) : datasetTrees
+      );
+      this.allStudies = this.searchKey(this.allStudyNodes[0][0], new Set<string>());
```

Comment:
This `this.allStudyNodes[0][0]` assumes a dataset that contains all studies. In other cases it breaks - this must be fixed.

Reply:
!